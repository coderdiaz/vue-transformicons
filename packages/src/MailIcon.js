import { BaseButton, IconMixin } from './BaseButton';

export default {
  mixins: [IconMixin],
  data() {
    return {
      defaultAnimation: 'mail',
      animations: {
        mail: ['tcon-mail--envelope'],
      },
    };
  },
  mounted() {
    this.defaultAnimation = (this.animation) ? this.animation : this.defaultAnimation;
  },
  render() {
    return <BaseButton
      ariaLabel="open mailbox"
      animation={this.defaultAnimation}
      animations={this.animations}
      active={this.active}>
      <span aria-hidden="true" class="tcon-mail--envelope__flap" />
      <span class="tcon-visuallyhidden">open mailbox</span>
    </BaseButton>;
  },
  components: {
    BaseButton,
  },
};
