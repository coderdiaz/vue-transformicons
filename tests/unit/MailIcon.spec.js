import { shallowMount } from '@vue/test-utils';
import MailIcon from '../../packages/src/MailIcon';

const vm = shallowMount(MailIcon);

// Helper function to create a component
const createComponent = propsData => shallowMount(MailIcon, { propsData });

describe('MailIcon', () => {
  let cmp;
  it('should match the snapshot', () => {
    expect(vm.$el).toMatchSnapshot();
  });

  it('has a mounted hook', () => {
    expect(typeof MailIcon.mounted).toBe('function');
  });

  describe('Properties', () => {
    it('should set the defaultAnimation if the prop animation isn\'t set', () => {
      cmp = createComponent();
      expect(cmp.vm.defaultAnimation).toBe('mail');
    });

    it('should set the default active value', () => {
      cmp = createComponent();
      expect(cmp.vm.active).toBeFalsy();
    });

    it('should set the prop active value', () => {
      cmp = createComponent();
      expect(cmp.vm.active).toBeFalsy();
      cmp.setProps({ active: true });
      expect(cmp.vm.active).toBeTruthy();
    });
  });
});
