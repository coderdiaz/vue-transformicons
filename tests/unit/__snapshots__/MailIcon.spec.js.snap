// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailIcon should match the snapshot 1`] = `undefined`;
